docs/devel/CONTRIBUTING.md

```diff
@@ -160,6 +160,71 @@ runtime installed. Currently supported runtime is `docker` only, You can run the
 $ make -C integration/ig/non-k8s test-docker
 ```
 
+### Benchmarks
+
+You can run the different benchmark tests with:
+
+```bash
+$ make gadgets-benchmarks
+```
+
+Or you can run an individual test with:
+
+```bash
+$ go test -exec sudo \
+    -bench='^BenchmarkAllGadgetsWithContainers$/^container10$/trace-tcpconnect' \
+    -run=Benchmark \
+    ./internal/benchmarks/...
+```
+
+Records of previous benchmarks are available [here](https://inspektor-gadget.github.io/ig-benchmarks/dev/bench/index.html).
+See details in the [CI documentation (benchmarks)](../ci.md#benchmarks).
+
+#### Explaining performance improvements in a PR
+
+If you want to contribute a performance improvement, it is useful to use benchmarks to explain the impact on
+performances. I will use the example of an improvement on the networking gadgets from
+[#1430](https://github.com/inspektor-gadget/inspektor-gadget/pull/1430):
+
+* Run the benchmarks both on the `main` and the feature branches and saving the output in two files. Please use at least `-count 8` to get usable results.
+```bash
+$ git checkout main
+$ go test -exec sudo \
+    -bench='^BenchmarkAllGadgetsWithContainers$/^container/trace-(dns|sni)' \
+    -run=Benchmark \
+    ./internal/benchmarks/... \
+    -count 8 | tee patched.bench
```

Comment:
From what I remember, it takes a few minutes with `-count 8`.

If you use too little, benchstat automatically tells you it cannot generate results and it advises to use `-count 6`. When the results are too unreliable, it automatically adds annotations to say which results are unreliable. In the output of this PR, there are no such annotations because it thinks it is good enough.

I could add some text to explain that.

Reply:
I see that the documentation now recommends `-count 10`. I'll update with that.

Reply:
It is not the means but the median.
I think it is enough to refer users to the benchstat documentation.

```
                                                  │ main.bench  │           patched.bench            │
                                                  │   sec/op    │   sec/op    vs base                │
AllGadgetsWithContainers/container100/trace-dns-4   2.941 ±  3%   1.489 ± 4%  -49.38% (p=0.000 n=10)
AllGadgetsWithContainers/container100/trace-sni-4   4.440 ± 19%   1.495 ± 6%  -66.34% (p=0.000 n=10)
geomean                                             3.613         1.492       -58.72%
```

In the dns benchmark, we have:
* main branch
  * the median (not the mean): 2.941 seconds per operation
  * the 95% confidence interval: ±  3%
* the feature branch
  * the median (not the mean): 1.489 seconds per operation
  * the 95% confidence interval: ±  4%
* comparing the two branches:
  * the feature branch reduces the latency by -49.38%
  * p-value of 0.000 (extremely unlikely that the difference between the medians was due to chance)
  * 10 samples